routes/Program/services.js

```diff
@@ -15,7 +29,127 @@ const getAll = async (query) => {
       .limit(query.size);
 };
 
+const getById = async (id, query) => {
+   return await Program.findById(id)
+       .skip(query.size * query.page)
+       .limit(query.size);
+};
+
+const getByName = async ( name, query) => {
```

Comment:
Replace with a single method that uses a composed condition. I mean, this implementation works as well, but is hard to maintain and exclusive conditions mean that for a new field that we can filter for, we will need just as many methods as we have already.

Reply:
done